api/librdb-ext-api.h

```diff
@@ -93,29 +93,20 @@ _LIBRDB_API RdbxFilterKey *RDBX_createHandlersFilterKey(RdbParser *p,
 typedef struct RdbxToRespConf {
     /* TODO: support the option of expire, del, select db */
 
+    int delKeyBeforeWrite;
+    int applySelectDbCmds;  /* if not configured SELECT commands, all keys will be flatten into a single db */
```

Comment:
Since it is already written & tested, maybe let's just leave it as is. WDYT?

Reply:
I guess we can leave it, but I agree with Oran it doesn't make much sense for a generic tool.

Reply:
Ok. I will drop this change.

Reply:
Reviewing again,I see now that this change requires to add a lot of redundant 
code to the tests which will also will make it less clear. That is, a very simple test 
such as testing SET command outputs, if until now it verified that the output is :
```
*3
$3
SET
$3
xxx
$3
111
```
Now it will verify:
```
*2
$6
SELECT
$1
0
*3
$3
SET
$3
xxx
$3
111
```
Please give it a second thought. Since there is no really additional complexity, easy to grasp, make tests simpler, and gives additional capability out of the box, I prefer to leave it as is.

Reply:
you can assume that the default database on connection is db 0, so you'll send SELECT only when changing to another db.

Reply:
IMHO, this is kind of hidden logic that can be much more risky to users who try to use or extend the parser, and say play around between db0 and db1, but they will end up with unclear scenario that sometimes it calls to SELECT, and sometimes not. Or worse ... wrongly uploading db_num_0.rdb on db 1.

Reply:
I don't think so, but what do you suggest?
A flag to omit SELECT that's off by default, but used in all tests?
IMHO always emitting SELECT, except for SELECT 0 when we're the ones that establish the connection and can assume db 0 is already selected is ok.

It gets complicated since we're not always the ones that establish the connection.
So maybe as an api, we can expose an optional argument for the caller to tell us which db is selected (or -1). And we keel track of what we changed. 
I.e if the input says db 4 is already selected, and the only db in the rdb is 4, we'll omit select.
And if the input is -1 we won't (even for db 0).

If the case we establish the connection, and also in tests, we can pass 0.

Reply:
The only reason i felt comfortable to have flag `applySelectDbCmds` in the API, is because behind 
the internal inscentive of using it for the tests, it suggests additional capability to the user. It opens 
the option to play the same RDB file on different db-nums. Or totally ignore the existance of 
db-nums in Redis.

Your last suggestion of passing selected db-num , offers nothing to the user. It just discard single
SELECT command if we are already on the same db-num. Which makes less sense to have it in 
the API.

Reply:
i suggested it for two reasons:
1. make the majority of the tests shorter
2. if we're establishing the connection, we can assume db0 is already selected.

i personally don't think we need to offer any such option to the user. if we keep it internal just for testing, then it's probably fine.
i personally don't think the tool should support flattering multi-db into one as a feature.
anyway, the discussion become too long, so do what you feel is right.

Reply:
ok. I removed the flag :). 

Instead, for testing purposes alone, I've defined custom filter handlers that do not pass new-db callbacks to the RDB to RESP handlers. Thanks.